shard_state_sync.go

```diff
@@ -42,8 +41,6 @@ type ShardStateSync interface {
 	End()
 	GetStartSequence(shardID *string) *string
 	Sync()
-}
-
-type ShardStateSyncOptions struct {
-	Ticker <-chan time.Time
+	TryAcquire(shardID *string) error
+	Release(shardID *string) error
```

Comment:
As we discussed, it's probably worth splitting these out to it's own interface/struct. We can compose them later if required. But they are doing somewhat different things.